Thumbnail/FormatThumbnail.php

```diff
@@ -84,18 +85,59 @@ public function generate(MediaProviderInterface $provider, MediaInterface $media
     /**
      * {@inheritdoc}
      */
-    public function delete(MediaProviderInterface $provider, MediaInterface $media)
+    public function delete(MediaProviderInterface $provider, MediaInterface $media, $format = null)
     {
-        // delete the differents formats
-        foreach ($provider->getFormats() as $format => $definition) {
-            $path = $provider->generatePrivateUrl($media, $format);
-            if ($path && $provider->getFilesystem()->has($path)) {
-                $provider->getFilesystem()->delete($path);
+        $filesystem = $provider->getFilesystem();
+
+        if (is_null($format)) {
+            //delete all thumbnails
+            $format = 'default_whateverformat';
+
+            $relPath = $provider->generatePrivateUrl($media, $format);
+            $relDir = dirname($relPath);
+            $fsRootDir = $this->getFilesystemRootDir($filesystem);
```

Comment:
Of course you can, there is no check on the format validity. The media and the format is enough to generate the private path, and so remove it from the fs driver.